src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIEnumerableConstructibleConverter.cs

```diff
@@ -10,33 +10,31 @@ namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class DefaultIEnumerableConstructibleConverter : JsonEnumerableConverter
     {
-        public static ConcurrentDictionary<Type, JsonPropertyInfo> s_objectJsonProperties = new ConcurrentDictionary<Type, JsonPropertyInfo>();
+        private static readonly ConcurrentDictionary<Type, JsonPropertyInfo> s_objectJsonProperties = new ConcurrentDictionary<Type, JsonPropertyInfo>();
```

Comment:
Since a `JsonPropertyInfo` instance contains state based on the options class (it looks at IgnoreNulls setting, etc plus more importantly the JsonPropertyInfo now holds onto a reference of the options class) this should not be a static. It should not be a static because the first options instance will define the state, and other option instances with different settings may not be compatible. Plus because it is static the options instance will never be GC'd even when the original options class is no longer used\referenced.

Instead, this cache should be located on the `JsonSerializerOptions` class and be non-static. However if there is a way to cache without using `JsonPropertyInfo`, or a way to pass `null` for the options instance, then it could remain a static.